examples/siro_sandbox/controllers/fetch_baselines_controller.py

```diff
@@ -410,7 +420,96 @@ def act(self, obs, env):
+            # There are two cases here. When the human is far away from the robot, we use
+            # normal point nav, and when the human is close enough, we switch to the
+            # social nav. This allows us to effectively find and follow the human
+            if self.should_start_skill:
+                human_robot_geo_dis = self._habitat_env._sim.geodesic_distance(
+                    robot_trans, human_trans
+                )
+
+                self.start_skill(obs, "nav_to_robot")
+
+            # First get the geo distance between the robot and the human and if the robot can see human
+            human_robot_geo_dis = self._habitat_env._sim.geodesic_distance(
+                robot_trans, human_trans
+            )
+            human_in_frame = self._batch_and_apply_transforms([obs])[
+                "humanoid_detector_sensor"
+            ]
+            if (
+                human_robot_geo_dis
+                > FOLLOW_SWITCH_GEO_DIS_FOR_POINT_SOCIAL_NAV
+                and not bool(human_in_frame)
+            ):
+                target_pddl_skill = "nav_to_obj"
+                # Robot is too far away from the human and not in frame, we should we point nav. Here we init the point nav policy
+                if (
+                    self._follow_state_policy_chosen == FOLLOWSTATEPOLICY.IDLE
+                    or self._follow_state_policy_chosen
+                    == FOLLOWSTATEPOLICY.SOCIAL_NAV
+                ):
+                    self.start_skill(obs, target_pddl_skill)
+                    self._follow_state_policy_chosen = (
+                        FOLLOWSTATEPOLICY.POINT_NAV
+                    )
+                type_of_skill = self.defined_skills.nav_to_obj.skill_name
+                max_skill_steps = (
+                    self.defined_skills.nav_to_obj.max_skill_steps
+                )
+            else:
+                target_pddl_skill = "nav_to_robot"
+                # Robot is close enough or robot can see human, we should we social nav. Here we init the social nav policy
+                if (
+                    self._follow_state_policy_chosen == FOLLOWSTATEPOLICY.IDLE
+                    or self._follow_state_policy_chosen
+                    == FOLLOWSTATEPOLICY.POINT_NAV
+                ):
+                    self.start_skill(obs, target_pddl_skill)
+                    self._follow_state_policy_chosen = (
+                        FOLLOWSTATEPOLICY.SOCIAL_NAV
+                    )
+                type_of_skill = self.defined_skills.nav_to_robot.skill_name
+                max_skill_steps = (
+                    self.defined_skills.nav_to_robot.max_skill_steps
+                )
+            step_terminate = self._skill_steps >= max_skill_steps
+
+            if type_of_skill == "OracleNavPolicy":
+                finished_nav = obs["agent_0_has_finished_oracle_nav"]
+            else:
+                # agent_trans = human_trans
```

Comment:
We probably want to remove this comment.

Reply:
Done, thanks!